src/main/java/org/traccar/database/NotificationManager.java

```diff
@@ -149,21 +155,30 @@ private void updateEvent(Event event, Position position) {
 
     private void forwardEvent(Event event, Position position) {
         if (eventForwarder != null) {
-            EventData eventData = new EventData();
-            eventData.setEvent(event);
-            eventData.setPosition(position);
-            eventData.setDevice(cacheManager.getObject(Device.class, event.getDeviceId()));
-            if (event.getGeofenceId() != 0) {
-                eventData.setGeofence(cacheManager.getObject(Geofence.class, event.getGeofenceId()));
-            }
-            if (event.getMaintenanceId() != 0) {
-                eventData.setMaintenance(cacheManager.getObject(Maintenance.class, event.getMaintenanceId()));
-            }
-            eventForwarder.forward(eventData, (success, throwable) -> {
-                if (!success) {
-                    LOGGER.warn("Event forwarding failed", throwable);
+            Device device = cacheManager.getObject(Device.class, event.getDeviceId());
+
+            Object allowForwarding = device.getAttributes().get(ATTRIBUTE_DEVICE_ALLOW_FORWARDING);
```

Comment:
Why is this an object? Use `getBoolean`.

Reply:
Changed it to 

```
boolean allowForwarding = (boolean) device.getAttributes().get(ATTRIBUTE_DEVICE_ALLOW_FORWARDING);
```